docs/data/tree-view/rich-tree-view/headless/headless.md

```diff
@@ -0,0 +1,279 @@
+---
+productId: x-tree-view
+title: Rich Tree View - Headless
+components: RichTreeView, TreeItem
+packageName: '@mui/x-tree-view'
+githubLabel: 'component: tree view'
+waiAria: https://www.w3.org/WAI/ARIA/apg/patterns/treeview/
+---
+
+# Rich Tree View - Headless
+
+<p class="description">Create your custom tree view.</p>
+
+:::warning
+The `useTreeView` hook is not public API for now,
+this page is still a _Work in Progress_ and its content is not meant to be used in production.
+:::
+
+## Create a custom plugin
+
+### Basic plugin
+
+A custom plugins contains 2 required elements:
+
+1. A hook that will be executed by the `useTreeView` hook:
+
+   ```ts
+   const useCustomPlugin = ({ params }) => {
+     React.useEffect(() => {
+       console.log(params.customParam);
```

Comment:
I honestly like the differentiation between `props` and `params` here. But I have nothing against renaming if you think it would make these concepts easier for users to understand 🤔